tasks/internal/get-advisory-severity/get-advisory-severity.yaml

```diff
@@ -114,22 +114,34 @@ spec:
           NUM_IMAGES=$(jq 'length' <<< "${IMAGES}")
           for ((i = 0; i < NUM_IMAGES; i++)); do
               image=$(jq -c --argjson i "$i" '.[$i]' <<< "${IMAGES}")
-              component=$(jq -r '.component' <<< "$image")
+              repository=$(jq -r '.repository' <<< "$image")
               NUM_CVES=$(jq '.cves.fixed | length' <<< "$image")
               for ((j = 0; j < NUM_CVES; j++)); do
                   cve=$(jq -r --argjson j "$j" '.cves.fixed | to_entries[$j].key' <<< "${image}")
-                  echo "Checking CVE ${cve} for component ${component}"
+                  echo "Checking CVE ${cve} for component with repository ${repository}"
                   # Get token. They are short lived, so get one for before each request
                   TOKEN=$(curl --retry 3 --negotiate -u : "${OSIDB_URL}"/auth/token | jq -r '.access')
                   OUTPUT=$(curl --retry 3 -H 'Content-Type: application/json' -H "Authorization: Bearer ${TOKEN}" \
                       "${OSIDB_URL}/osidb/api/v1/flaws?cve_id=${cve}&include_fields=${INCLUDE_FIELDS}" \
                       | jq '.results[0]')
                   IMPACT=$(jq -r '.impact' <<< "$OUTPUT")
                   # If there is a component specific impact, use that instead
-                  if PS_COMP=$(jq -e --arg comp "$component" '.affects[] | select(.ps_component==$comp)' <<< "$OUTPUT")
-                  then
-                      IMPACT=$(jq -r --arg default "$IMPACT" '.impact // $default' <<< "$PS_COMP")
-                  fi
+                  # To check if it is the same component, we use the repository value and match it with the
+                  # repository_url field in the purl from OSIDB. Thus, we have to check each entry as
+                  # repository_url is not its own field
+                  NUM_AFFECTED_COMPONENTS=$(jq '.affects | length' <<< "$OUTPUT")
+                  for ((k = 0; k < NUM_AFFECTED_COMPONENTS; k++)); do
+                      AFFECTED_COMPONENT=$(jq --argjson k "$k" '.affects[$k]' <<< "$OUTPUT")
+                      PURL=$(jq -r '.purl // ""' <<< "$AFFECTED_COMPONENT")
+                      # .purl can be empty, so we direct stderr and just have it return empty string if purl was empty
+                      AFFECTED_REPOSITORY=$(python3 -c "from packageurl import PackageURL; \
+                        print(PackageURL.from_string('$PURL').to_dict()['qualifiers']['repository_url'])" \
```

Comment:
I did not see the `qualifiers` parameter in the "results" string. Is is optional in the URL? If yes should the test cover both cases - with and without the qualifiers parameter. 
 
 I wonder if we should have a function to parse  the URL (scheme, host and query string) instead of importing external packages. WDYT? It might be useful in the future - or not.

Reply:
We have lots of purl examples here
https://github.com/konflux-ci/release-service-utils/blob/main/pyxis/test_upload_rpm_data.py#L22
There is no "qualifiers" parameter in the string. It is just how the function works. I am not trying to cover all test cases for the `PackageURL.from_string()` function as we don't maintain it, that's why I don't have test cases like you describe. If it works, I use it. Otherwise, it is the empty string.

As far as a function to parse it, I would agree if I was doing at least a couple steps. I am literally just calling `PackageURL.from_string` though. I don't think we should maintain our own version of what is just an externally maintained function with nothing special added on top

Reply:
understood. My concern with the package was if for any reason the package is gone and the import does not work - which anyhow would be an easy fix.  LGTM then.